Assets/Scripts/Entity/Player/Possession.cs

```diff
@@ -76,19 +78,45 @@ void UnHighLightEnemy()
         HaloActive = false;
         targetedEnemy = null;
     }
-
+    
     //Actual possession takes place
     void TakePossessionAction(GameObject enemy){
         
         if(enemy == null)
         return;
         
+        double heartdiff = (player.Health - enemy.GetComponent<Enemy>().MaxHealth) / 2;
```

Comment:
All of this logic is so confusing and obtuse... but at this point, I guess there is no point in caring too much.